library/alloc/src/sync.rs

```diff
@@ -804,6 +804,7 @@ impl<T: ?Sized> Arc<T> {
     /// let x_ptr = Arc::into_raw(x);
     /// assert_eq!(unsafe { &*x_ptr }, "hello");
     /// ```
+    #[must_use = "losing the pointer will leak memory"]
```

Comment:
This one feels like it could lead to false positives. This is Arc, so you *can* get the data back via another instance.

not sure if it happens in practice though. I *think* I used that once to simulate 
increment_strong_count? Not sure.